### PR TITLE
Added `rewrite(Integral)` to elliptic_integrals

### PR DESCRIPTION
**Pull request by czgdp1807**
#### References to other Issues or PRs

Closes #12136 

#### Release Notes

* functions
  * `rewrite(Integral)` has been added to `sympy.functions.special.elliptic_integrals`

### CODE REVIEW

**Comment by czgdp1807** [comment]
ping @cbm755 @oscarbenjamin

**Comment by czgdp1807** [comment]
The PR is almost ready, once the tensorflow fix is done and reviews are received, we will merge it.

**Comment by czgdp1807** [comment]
As everything looks fine with this PR, we will merge it within few hours, if no critical objections are raised.

**Comment by jksuom** [comment]
Elliptic integrals have two common normal forms. This is the trigonometric form. Should we also have the Legendre normal form?

**Comment by czgdp1807** [comment]
How should we allow the user to pass a parameter to select from one of the two? Any suggestions? I think it should be done using `rewrite` only. What about allowing a keyword argument in `rewrite`?

**Comment by czgdp1807** [comment]
And should we do that in this PR or a separate one?

**Comment by jksuom** [comment]
I believe that rewrite will accept keyword arguments. That could be `trig`, for example.

**Comment by czgdp1807** [comment]
@jksuom Can you please share some links on trigonometric forms and Legendre normal forms of elliptic integrals?

**Comment by jksuom** [comment]
https://en.wikipedia.org/wiki/Elliptic_integral

File: sympy/functions/special/elliptic_integrals.py [hunk 1]
```diff
@@ -412,3 +430,15 @@ def fdiff(self, argindex=1):
...
+
+    def _eval_rewrite_as_Integral(self, *args):
+        from sympy import Integral, Dummy
+        n, m, z = self.args[0], self.args[1], None
+        if len(self.args) == 3:
+            z = self.args[2]
+        if z is None:
+            z = pi/2
+        else:
+            n, z, m = n, m, z
```
Comment (cbm755) [inline sympy/functions/special/elliptic_integrals.py]: Is this the same as:
```py
if len(self.args) == 2:
  n, m, z = self.args[0], self.args[1], pi/2
else:
  n, z, m = self.args
```

File: sympy/functions/special/tests/test_elliptic_integrals.py [hunk 2]
```diff
@@ -150,3 +159,8 @@ def test_P():
 
     assert P(n, z, m).series(z) == z + z**3*(m/6 + n/3) + \
         z**5*(3*m**2/40 + m*n/10 - m/30 + n**2/5 - n/15) + O(z**6)
+
+    assert P(n, z, m).rewrite(Integral).dummy_eq(
+        Integral(1/((1 - n*sin(t)**2)*sqrt(1 - m*sin(t)**2)), (t, 0, z)))
+    assert P(n, m).rewrite(Integral).dummy_eq(
+        Integral(1/((1 - n*sin(t)**2)*sqrt(1 - m*sin(t)**2)), (t, 0, pi/2)))
```
Comment (cbm755) [inline sympy/functions/special/tests/test_elliptic_integrals.py]: I like this use of `.dummy_eq` better than using `str`.

**Comment by cbm755** [comment]
thanks for continuing this work @czgdp1807.  This looks good to me.  I have no strong opinion on whether Legendre ve forms should be in this PR or a new one, although I'd lean towards just merging this and doing the alternative form in a new PR.

(I have no opinion at the moment which should be the default!)

**Comment by czgdp1807** [comment]
Once the tests pass I will merge it.

**Comment by oscarbenjamin** [comment]
No objection to this PR but an observation:

Every method of this class branches on `len(self.args)`. This should be split into two classes.

**Pull request merged**